lighthouse-core/test/audits/uses-rel-preload-test.js

```diff
@@ -9,56 +9,105 @@
 /* eslint-env mocha */
 
 const UsesRelPreload = require('../../audits/uses-rel-preload.js');
+const NetworkNode = require('../../lib/dependency-graph/network-node');
 const assert = require('assert');
 const defaultMainResource = {
   _endTime: 1,
 };
 
-const mockArtifacts = (networkRecords, mockChain, mainResource = defaultMainResource) => {
-  return {
-    devtoolsLogs: {
-      [UsesRelPreload.DEFAULT_PASS]: [],
-    },
-    requestCriticalRequestChains: () => {
-      return Promise.resolve(mockChain);
-    },
-    requestNetworkRecords: () => networkRecords,
-    requestMainResource: () => {
-      return Promise.resolve(mainResource);
-    },
+describe('Performance: uses-rel-preload audit', () => {
+  let mockGraph;
+  let mockSimulator;
+
+  const mockArtifacts = (networkRecords, mockChain, mainResource = defaultMainResource) => {
+    return {
+      traces: {[UsesRelPreload.DEFAULT_PASS]: {traceEvents: []}},
+      devtoolsLogs: {[UsesRelPreload.DEFAULT_PASS]: []},
+      requestCriticalRequestChains: () => {
+        return Promise.resolve(mockChain);
+      },
+      requestLoadSimulator: () => mockSimulator,
+      requestPageDependencyGraph: () => mockGraph,
+      requestNetworkRecords: () => networkRecords,
+      requestMainResource: () => {
+        return Promise.resolve(mainResource);
+      },
+    };
   };
-};
 
-describe('Performance: uses-rel-preload audit', () => {
-  it(`should suggest preload resource`, () => {
+  function buildNode(requestId, url) {
+    return new NetworkNode({url, requestId});
+  }
+
+  it('should suggest preload resource', () => {
+    const rootNode = buildNode(1, 'http://example.com');
+    const mainDocumentNode = buildNode(2, 'http://www.example.com');
+    const scriptNode = buildNode(3, 'http://www.example.com/script.js');
+    const scriptAddedNode = buildNode(4, 'http://www.example.com/script-added.js');
+
+    mainDocumentNode.addDependency(rootNode);
+    scriptNode.addDependency(mainDocumentNode);
+    scriptAddedNode.addDependency(scriptNode);
+
+    mockGraph = rootNode;
+    mockSimulator = {
```

Comment:
is this reused for the other tests?

Reply:
nope, just scoped for the `mockArtifacts` to be able to return it

Reply:
wait, but how are the others running with an undefined simulator? (obvs I'm too lazy to debug here :)

Reply:
the other tests are asserting that *no* savings are found, so a simulator isn't necessary

Reply:
oh, lol